metaflow/multicore_utils.py

```diff
@@ -68,8 +69,17 @@ def parallel_imap_unordered(func, iterable, max_parallel=None, dir=None):
 
     while pids:
 
-        pid, output_file = pids.pop()
-        if os.waitpid(pid, 0)[1]:
+        for idx, pid_info in enumerate(pids):
+            pid, output_file = pid_info
+            pid, exit_code = os.waitpid(pid, os.WNOHANG)
+            if pid:
+                pids.pop(idx)
+                break
+        else:
+            time.sleep(1)  # Wait a bit before re-checking
```

Comment:
After a quick discussion, someone brought up the fact that for very short tasks, this now makes it at least 1 second. Could we either skip the sleep (and just do tight polling) or do something a bit fancier (no sleep initially and then some sleep after a while)?

Reply:
I did consider that as well but decided not to optimize as it might be premature and I am not sure if those very short tasks (less than 1 sec) would need to use this function?  If you are certain there are real scenarios where all tasks are done less than 1 sec and they would use this multi-process function for that, then I can optimize it.

Reply:
Reduced sleep to 0.1 secs as that should be small enough for any short tasks while not affecting the CPU as it checks.